### PR TITLE
Governor: Proposal Creation, Voting, Proposal Queuing, Proposal Execution

### PR DESCRIPTION
**Pull request by TravellerOnTheRun**

### CODE REVIEW

File: contracts/Governor.sol [hunk 1]
```diff
@@ -29,21 +29,21 @@ contract RootDao is
...
-    __Ownable_init(initialOwner);
-    __UUPSUpgradeable_init();
-  }
+    function initialize(
+        IVotes voteToken, 
+        TimelockControllerUpgradeable timelockController, 
+        address initialOwner
+    ) public initializer {
+        __Governor_init("RootDao");
+        __GovernorSettings_init(1 /* 1 day */, 60 /* 1 week */, 10 * 10 ** 18);
```
Comment (cybervoid0) [inline contracts/Governor.sol]:
```suggestion
        __GovernorSettings_init(1 /* 1 block */, 60 /* 30 minutes */, 10 * 10 ** 18);
```

**Review by cybervoid0 — APPROVED** [review]

File: test/Governor.test.ts [hunk 2]
```diff
@@ -57,22 +69,51 @@ describe('RootDAO Contact', () => {
...
+    const createProposal = async (proposalDesc = defaultDescription) => {
+      const blockHeight = await ethers.provider.getBlockNumber()
+      const votingDelay = await governor.votingDelay()
+
+      const calldata = stRIF.interface.encodeFunctionData('symbol')
+      proposal = [[await stRIF.getAddress()], [0n], [calldata]]
+
+      proposalId = await governor
+        .connect(holders[0])
+        .hashProposal(proposal[0], proposal[1], proposal[2], generateDescriptionHash(proposalDesc))
```
Comment (cybervoid0) [inline test/Governor.test.ts]:
```suggestion
        .hashProposal(...proposal, generateDescriptionHash(proposalDesc))
```

**Review by cybervoid0 — APPROVED** [review]

File: scripts/deploy-governor.ts [hunk 3]
```diff
@@ -1,29 +1,47 @@
...
+
+  // grant Executor role to the Governor
+  const executorRole = await timelock.EXECUTOR_ROLE()
+  const grantExecutorRoleTx = await timelock.grantRole(executorRole, governorAddr)
+  await grantExecutorRoleTx.wait()
 
-  const rootDAOContact = await rootDAOGovernor.waitForDeployment()
+  // renounce Admin role - this operation should be finally done by the DAO deployer
+  const adminRole = await timelock.DEFAULT_ADMIN_ROLE()
+  const [deployer] = await hre.ethers.getSigners()
```
Comment (sleyter93) [inline scripts/deploy-governor.ts]:
```suggestion
  const [deployer] = await ethers.getSigners()
```

File: scripts/deploy-governor.ts [hunk 4]
```diff
@@ -1,29 +1,47 @@
...
-import { GovernorTimelockControlUpgradeable, RootDao } from '../typechain-types'
+import { RootDao } from '../typechain-types'
 
-export const deployGovernor = async (tokenAddress: string, deployerAddress: string) => {
+export const deployGovernor = async (
+  tokenAddress: string,
+  deployerAddress: string,
+  timelockAddress: string,
+) => {
+  const timelock = await hre.ethers.getContractAt('DaoTimelockUpgradable', timelockAddress)
```
Comment (sleyter93) [inline scripts/deploy-governor.ts]:
```suggestion
  const timelock = await ethers.getContractAt('DaoTimelockUpgradable', timelockAddress)
```

File: scripts/deploy-timelock.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    initializer: 'initialize',
+    kind: 'uups',
+    timeout: 0,
+  })
+  await timelock.waitForDeployment()
+
+  console.log(`Deployed Timelock on ${hre.network.name} with address ${await timelock.getAddress()}`)
+
+  return timelock as unknown as DaoTimelockUpgradable
+}
```
Comment (sleyter93) [inline scripts/deploy-timelock.ts:29]: Maybe we need to invoke the `deployTimeLock` function like in deploy-tokens.ts

**Review by sleyter93 — COMMENTED** [review]
Good Job!
One last thing, given we have some scripts with ethers and viem. To run the coverage, we need to execute the following command: `SOLIDITY_COVERAGE=true npx hardhat coverage`. Let's include this in the README.

**Pull request merged**